src/validate.ts

```diff
@@ -20,54 +17,35 @@ const ON_OFF_STATES_SCHEMA = require('../traits/onoff.states.schema.json');
  */
 function responseValidation(apiResponse: object, schema: object) {
   const isValid = ajv.validate(schema, apiResponse);
-  try {
-    if (isValid) {
-      return undefined;
-    } else {
-      throw new Error();
-    }
-  } catch (e) {
+  if (isValid) {
+    return undefined;
+  } else {
     return ajv.errors;
   }
 }
+
 /**
  * Identifies the response type and validates the function based on the schemas.
- * @param apiResponse User defined api response
- * @param responseType User defined intent response
+ * @param req Request from one of the generate request functions.
+ * @param apiResponse User defined api response.
  * @return Errors from AJV validation, if any. Undefined otherwise.
  */
-export function validate(apiResponse: object, responseType: string) {
-  if (responseType === 'sync') {
-    // Sync response validations
-    const DEVICES = apiResponse['payload']['devices'];
-    const DEVICES_LENGTH = apiResponse['payload']['devices'].length;
-    for (let i = 0; i <= DEVICES_LENGTH; i++){
-        const DEVICE_TRAITS = DEVICES[i]['traits'];
-        const TRAITS_LENGTH = DEVICE_TRAITS.length;
-        const DEVICE_ATTRIBUTES = DEVICES[i]['attributes'];
-        for (let j = 0; j <= TRAITS_LENGTH; j++){
-            //validate this against the attributes
-            const DEVICE_TRAITS[j]
-        }
-    } 
+export function validate(req: object, apiResponse: object) {
+  const responseType = req['inputs'][0]['intent'];
+  if (responseType === 'action.devices.SYNC') {
     return responseValidation(apiResponse, SYNC_RESPONSE_SCHEMA);
-  } else if (responseType == 'query') {
-    // Query response validations
-        // loop using the device data given from generate requests 
-    // validate against the onoff params schema here
+  } else if (responseType == 'action.devices.QUERY') {
     return responseValidation(apiResponse, QUERY_RESPONSE_SCHEMA);
-  } else if (responseType == 'execute') {
-    // Execute response validations
-    const COMMANDS = apiResponse['payload']['commands'];
-    const COMMANDS_LENGTH = COMMANDS.length;
-    for (let i = 0; i <= COMMANDS_LENGTH; i++) {
-      const STATES = COMMANDS[i]['states'];
-      return responseValidation(STATES, ON_OFF_STATES_SCHEMA);
+  } else if (responseType == 'action.devices.EXECUTE') {
+    const COMMANDS = EXECUTE_RESPONSE_SCHEMA["payload"]["commands"];
+    const COMMANDS_LENGTH = COMMANDS.length;     
+    for(let i = 0; i <= COMMANDS_LENGTH; i++){
+        const IS_ON = COMMANDS[i]["states"]["on"];
+        const IS_ONLINE = COMMANDS[i]["states"]["online"];
+        return [responseValidation(IS_ON, ON_OFF_STATES_SCHEMA), responseValidation(IS_ONLINE, ON_OFF_STATES_SCHEMA)];
```

Comment:
You may want to consider the best return type. If we start returning in an array, how does that affect the developer experience and tests?